src/matrixToDataFrame.c

```diff
@@ -0,0 +1,312 @@
+#include <assert.h>
+#include <inttypes.h>
+#include <limits.h>
+#include <stdint.h>
+#include <string.h>
+#include "tibble.h"
+
+static SEXP pairlist_shallow_copy(SEXP p)
+{
+    SEXP attr, q, s;
+    int nprot = 0;
+
+    PROTECT(attr = Rf_cons(CAR(p), R_NilValue)); nprot++;
+    q = attr;
+    SET_TAG(q, TAG(p));
+    p = CDR(p);
+
+    while (!Rf_isNull(p)) {
+        PROTECT(s = Rf_cons(CAR(p), R_NilValue)); nprot++;
+        SETCDR(q, s);
+        q = CDR(q);
+        SET_TAG(q, TAG(p));
+        p = CDR(p);
+    }
+
+    UNPROTECT(nprot);
+    return attr ;
+}
+
+static void copy_attributes(SEXP out, SEXP data)
+{
+    SEXP att = ATTRIB(data);
+    if (!Rf_isNull(att)){
+        SET_ATTRIB(out, pairlist_shallow_copy(ATTRIB(data))) ;
+    }
+    SET_OBJECT(out, OBJECT(data));
+    if (IS_S4_OBJECT(data)) {
+        SET_S4_OBJECT(out);
+    }
+}
+
+// same as copy_attributes but without names
+static void copy_most_attributes(SEXP out, SEXP data)
+{
+    copy_attributes(out, data);
+    Rf_setAttrib(out, R_NamesSymbol, R_NilValue);
+}
+
+static void get_dim(SEXP x, R_xlen_t *nrowptr, R_xlen_t *ncolptr)
+{
+    SEXP dim;
+    R_xlen_t nrow = 0, ncol = 0;
+    int nprot = 0;
+
+    PROTECT(dim = getAttrib(x, R_DimSymbol)); nprot++;
```

Comment:
Can we use `Rf_getAttrib()` instead of `getAttrib()` here, to make the use of the R API more obvious? Also for the other functions? Please define `R_NO_REMAP` to turn off availability of functions without `Rf_` prefix.